apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TableSchemaBuilder.cs

```diff
@@ -32,6 +32,7 @@ namespace Google.Cloud.BigQuery.V2
     public sealed class TableSchemaBuilder : IEnumerable
     {
         private readonly List<TableFieldSchema> _fields = new List<TableFieldSchema>();
+        private readonly Dictionary<string, TableFieldSchema> _tableFields = new Dictionary<string, TableFieldSchema>();
```

Comment:
I wouldn't bother building this dictionary. I'd just use `_fields.FirstOrDefault(field => field.Name == fieldname)` to fetch the field within `ModifyField`. Having parallel collections generally makes it slightly trickier to maintain... and my proposed solution also correctly handles weirdnesses such as:

```csharp
var schema = new TableSchemaBuilder
{
    { "name1", ... }
}.ModifyField("name1", field => field.Name = "name2")
.ModifyField("name2", field => field.Name = "name3")
.Build();
```

Reply:
haven't considered that thanks for point this out rectified.